File: 151051/Goron/redis-task/discussion-service/src/main/java/by/bsuir/discussionservice/entity/Message.java
package by.bsuir.discussionservice.entity;

import java.io.Serializable;

import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyClass;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Table("tbl_message")
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class Message {

    @PrimaryKeyClass
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    @Getter
    @Setter
    public static class Key implements Serializable  {
        @PrimaryKeyColumn(ordinal = 0, type = PrimaryKeyType.PARTITIONED)
        private String country;
        
        @PrimaryKeyColumn(ordinal = 1, type = PrimaryKeyType.CLUSTERED)
        private Long id;
        
        @PrimaryKeyColumn(name = "news_id", ordinal = 2, type = PrimaryKeyType.CLUSTERED)
        private Long newsId;
    }

    @PrimaryKey
    private Key key;

    @Column("content")
    private String content;

    @Column("state")
    private MessageState state;
}

File: 251004/Asepkov/internal/discussion/model/message.go
package model

type MessageState string

const (
	StatePending MessageState = "PENDING"
	StateApprove MessageState = "APPROVE"
	StateDecline MessageState = "DECLINE"
)

// Message represents a discussion message in the system
type Message struct {
	ID      int64        `json:"id" cql:"id"`
	Country string       `json:"country" cql:"country"`
	NewsID  int64        `json:"newsId" cql:"newsid"`
	Content string       `json:"content" cql:"content"`
	State   MessageState `json:"state"`
}

// MessageTable represents the Cassandra table name for messages
const MessageTable = "tbl_message"
